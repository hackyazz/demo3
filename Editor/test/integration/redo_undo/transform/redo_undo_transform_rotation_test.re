open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: transform rotation", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode,
      );
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe
    ("fix bug",
    (
    () => {
    /* describe("test undo operate", () =>
      describe("test undo one step", () =>
        test(
          {|
            set current scene tree node to c1;
            change c1->local euler angles;
            undo;

            ui->inspector->transform->c1->local euler angles should be (0,0,0);
            |},
          () => {
            let currentGameObjectTransform =
              GameObjectTool.getCurrentSceneTreeNodeTransform();

            MainEditorTransformTool.changeRotationYAndBlur(
              ~transform=currentGameObjectTransform,
              ~value=10.0,
              (),
            );
            RedoUndoTool.undoHistoryState();

            BuildComponentForCurryTool.buildMainEditorTransformComponent()
            |> ReactTestTool.createSnapshotAndMatch;
          },
        )
      )
    ); */

        test(
          {|
            set current scene tree node to c1;
            change c1->local euler angles;
            undo;

            ui->inspector->transform->c1->local euler angles should be (0,0,0);
            |},
          () => {
            let currentGameObjectTransform =
              GameObjectTool.getCurrentSceneTreeNodeTransform();

            MainEditorTransformTool.changeRotationYAndBlur(
              ~transform=currentGameObjectTransform,
              ~value=10.0,
              (),
            );
            RedoUndoTool.undoHistoryState();

            BuildComponentForCurryTool.buildMainEditorTransformComponent()
            |> ReactTestTool.createSnapshotAndMatch;
          },
        );
        test(
          {|
            set current scene tree node to c1;
            change c1->local euler angles;
            undo;

            ui->inspector->transform->c1->local euler angles should be (0,0,0);
            |},
          () => {
            let currentGameObjectTransform =
              GameObjectTool.getCurrentSceneTreeNodeTransform();

            MainEditorTransformTool.changeRotationYAndBlur(
              ~transform=currentGameObjectTransform,
              ~value=10.0,
              (),
            );
            RedoUndoTool.undoHistoryState();

            BuildComponentForCurryTool.buildMainEditorTransformComponent()
            |> ReactTestTool.createSnapshotAndMatch;
          },
        )

    })
    );

    
  });