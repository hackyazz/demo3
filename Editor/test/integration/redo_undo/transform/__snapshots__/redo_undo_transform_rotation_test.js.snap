// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: transform rotation fix bug 
            set current scene tree node to c1;
            change c1->local euler angles;
            undo;

            ui->inspector->transform->c1->local euler angles should be (0,0,0);
             1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`redo_undo: transform rotation fix bug 
            set current scene tree node to c1;
            change c1->local euler angles;
            undo;

            ui->inspector->transform->c1->local euler angles should be (0,0,0);
             2`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
